portlist/portlist_windows.go

```diff
@@ -52,6 +53,8 @@ func (im *windowsImpl) AppendListeningPorts(base []Port) ([]Port, error) {
 		return nil, err
 	}
 
+	pi := newPidInfo()
```

Comment:
let's do this lazily when it's first needed. Most calls (every second or so) there will be no changes in the set of open TCP connections so this work would be otherwise wasted and never used. (the line 78 continue would hit each time)

Reply:
I realized later that it might not've been obvious how often this function (AppendListeningPorts) runs. On Linux it's every 1 second. On Windows it's been 5 seconds because of its relative speed. Ideally it'd be nice to be 1 second also. 

Actually, ideally we'd not poll every N seconds at all on any platform and instead just subscribe (on Linux: via eBPF) to changes in the TCP/UDP listening tables.

But meanwhile assume that this runs every 5 seconds.